File: figma-mcp-server-python/explore_figma.py
#!/usr/bin/env python3

"""
Fetch Figma design structure to find POC designs
"""

import os
import json
import requests
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

FIGMA_API_TOKEN = os.getenv("FIGMA_API_TOKEN")
FILE_KEY = 'yFsZrVGKYZvCnaL7AcnU8n'

if not FIGMA_API_TOKEN:
    print('Error: FIGMA_API_TOKEN environment variable is not set')
    exit(1)

def list_nodes(nodes, depth=0):
    """Recursively list all frames and components."""
    if not nodes:
        return

    for node in nodes:
        indent = '  ' * depth
        if node.get('type') == 'COMPONENT':
            icon = '🧩'
        elif node.get('type') == 'FRAME':
            icon = '📦'
        else:
            icon = '📄'
        node_id = f", id: {node['id']}" if node.get('id') else ''
        print(f"{indent}{icon} {node['name']} ({node['type']}{node_id})")

        if node.get('children') and len(node['children']) > 0:
            list_nodes(node['children'], depth + 1)

def fetch_figma_file():
    try:
        print('📥 Fetching Figma file structure...\n')

        response = requests.get(
            f"https://api.figma.com/v1/files/{FILE_KEY}",
            headers={"X-Figma-Token": FIGMA_API_TOKEN}
        )
        response.raise_for_status()

        data = response.json()

        print('📊 Figma File Structure:\n')
        if data.get('document') and data['document'].get('children'):
            list_nodes(data['document']['children'])

        # Save full data to file
        with open('figma-structure.json', 'w') as f:
            json.dump(data, f, indent=2)

        print('\n💾 Full data saved to figma-structure.json')

    except requests.RequestException as e:
        print(f'❌ Error: {e}')
        exit(1)

if __name__ == "__main__":
    fetch_figma_file()

File: figma-mcp-server/explore-figma.mjs
#!/usr/bin/env node

/**
 * Fetch Figma design structure to find POC designs
 */

import fetch from 'node-fetch';

const FIGMA_API_TOKEN = process.env.FIGMA_API_TOKEN;
const FILE_KEY = process.argv[2] || process.env.FIGMA_FILE_KEY;

if (!FIGMA_API_TOKEN) {
  console.error('Error: FIGMA_API_TOKEN environment variable is not set');
  process.exit(1);
}

if (!FILE_KEY) {
  console.error('Error: Figma file key is required. Pass it as the first argument or set FIGMA_FILE_KEY.');
  process.exit(1);
}

async function fetchFigmaFile() {
  try {
    console.log('📥 Fetching Figma file structure...\n');
    
    const response = await fetch(`https://api.figma.com/v1/files/${FILE_KEY}`, {
      headers: {
        'X-Figma-Token': FIGMA_API_TOKEN,
      },
    });

    if (!response.ok) {
      throw new Error(`Figma API error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    
    // List all frames and components
    function listNodes(nodes, depth = 0) {
      if (!nodes) return;
      
      for (const node of nodes) {
        const indent = '  '.repeat(depth);
        const icon = node.type === 'COMPONENT' ? '🧩' : node.type === 'FRAME' ? '📦' : '📄';
        console.log(`${indent}${icon} ${node.name} (${node.type}${node.id ? `, id: ${node.id}` : ''})`);
        
        if (node.children && node.children.length > 0) {
          listNodes(node.children, depth + 1);
        }
      }
    }

    console.log('📊 Figma File Structure:\n');
    if (data.document && data.document.children) {
      listNodes(data.document.children);
    }
    
    console.log('\n💾 Full data saved to figma-structure.json');
    
  } catch (error) {
    console.error('❌ Error:', error);
    process.exit(1);
  }
}

fetchFigmaFile();
